### PR TITLE
Rust 1.15.1

### PR DESCRIPTION
**Pull request by brson**
r? @steveklabnik 

Don't merge.

### CODE REVIEW

File: _posts/2017-02-08-Rust-1.15.1.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+With that, proper uniqueness invariants are maintained, only one mutable slice
+can be created at a time, and `v_iter` must be declared mutable in order to pull
+out a mutable slice.
+
+So we made that change, and we're releasing a fix. Since this is a new, minor
+feature, and the present implementation is spectacularly unsound, we decided to
+go ahead and release the fix immediately, hopefully before too many codebases
+pick it up - that is, we don't consider this a breaking change that requires a
+careful transition, but a necessary bug fix.
```
Comment (steveklabnik) [inline _posts/2017-02-08-Rust-1.15.1.md]: I think citing our RFCs here is a good idea

**Comment by brson** [comment]
[www](https://github.com/rust-lang/rust-www/pull/722)

File: _posts/2017-02-08-Rust-1.15.1.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+So we made that change, and we're releasing a fix. In Rust we take pride in not
+breaking APIs, but since this is a new, minor feature, and the present
+implementation is spectacularly unsound, we decided to go ahead and release the
+fix immediately, hopefully before too many codebases pick it up - that is, we
+don't consider this a breaking change that requires a careful transition, but a
+necessary bug fix. For more about Rust's approach to ensuring stability see the
+["Stability as a Deliverable"][stab] blog post, [RFC 1122], on language
+evolution, and [RFC 1105], on library evolution (curiously, RFC 1105 does not
+actually contain any language allowing for library breakage due to soundness,
```
Comment (steveklabnik) [inline _posts/2017-02-08-Rust-1.15.1.md]: 😓

File: _posts/2017-02-08-Rust-1.15.1.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+to ensure patches that touch unsafe blocks get more scrutiny during review: this
+patch was written by an expert Rust programmer, reviewed by an expert Rust
+programmer, but it appeared trivial and admittedly was not scrutinized closely.
+In this we could learn from our sister project, [Servo], which uses a bot to
+alert reviewers to pull requests that touch unsafe blocks. There are other ideas
+for validating unsafe blocks in Rust that are more thorough, from human-language
+attestation that the block is correct to bolt-on theorem provers. At the moment
+the Rust library team doesn't have plans to make process changes based on this
+mistake, but unsafe validation in Rust will be an area of active research
+forever, and we'll continue to evaluate our options.
```
Comment (steveklabnik) [inline _posts/2017-02-08-Rust-1.15.1.md]: So, this was where you said it felt awkward. I agree. I think maybe cutting out the ideas is better; keep it succinct. 

> What can we learn from this? Well, it's at least good to be reminded why we endeavored to write a safe language in the first place: writing safe code is not easy. We have some ideas for how to catch this kind of problem earlier in the development process, but we haven't made any decisions yet.

WDYT?

**Comment by brson** [comment]
Updated with your language @steveklabnik

**Comment by alexcrichton** [comment]
I've updated this with an anticipated 2017-02-09 release date

File: _posts/2017-02-09-Rust-1.15.1.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+This release fixes two issues, a soundness bug in the new
+`vec::IntoIter::as_mut_slice` method, and a regression wherein certain C
+components of the Rust distribution were [not compiled with `-fPIC`][fpic].  The
+latter results in the text section of executables being writable in some
+configurations, including common Linux configurations, subverting an important
+attack mitigation, and causing longer startup times by causing the linker to do
+more work. For mostly-Rust codebases, the practical impact of losing read-only
+text sections is relatively small (since Rust's type system is its first line of
+defense), but for Rust linked into other codebases the impact could be
+unexpectedly quite significant. The details of the bug are not that interesting
```
Comment (steveklabnik) [inline _posts/2017-02-09-Rust-1.15.1.md]: I would say

> The impact of not compiling with `-fPIC` is well-understood, so the rest of this post focuses on the soundness bug.

File: _posts/2017-02-09-Rust-1.15.1.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+[solution]: https://github.com/rust-lang/rust/pull/39466
+
+With that, proper uniqueness invariants are maintained, only one mutable slice
+can be created at a time, and `v_iter` must be declared mutable in order to pull
+out a mutable slice.
+
+So we made that change, and we're releasing a fix. In Rust we take pride in not
+breaking APIs, but since this is a new, minor feature, and the present
+implementation is spectacularly unsound, we decided to go ahead and release the
+fix immediately, hopefully before too many codebases pick it up - that is, we
```
Comment (steveklabnik) [inline _posts/2017-02-09-Rust-1.15.1.md]: —

File: _posts/2017-02-09-Rust-1.15.1.md [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+don't consider this a breaking change that requires a careful transition, but a
+necessary bug fix. For more about Rust's approach to ensuring stability see the
+["Stability as a Deliverable"][stab] blog post, [RFC 1122], on language
+evolution, and [RFC 1105], on library evolution (curiously, RFC 1105 does not
```
Comment (steveklabnik) [inline _posts/2017-02-09-Rust-1.15.1.md]: i am tempted to remove this parenthetical entirely

Comment (brson) [inline _posts/2017-02-09-Rust-1.15.1.md]: agree

Comment (brson) [inline _posts/2017-02-09-Rust-1.15.1.md]: it's pretty far in the weeds

File: _posts/2017-02-09-Rust-1.15.1.md [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+[notes]: https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1151-2017-02-09
+
+### What's in 1.15.1 stable
+
+This release fixes two issues, a soundness bug in the new
+`vec::IntoIter::as_mut_slice` method, and a regression wherein certain C
+components of the Rust distribution were [not compiled with `-fPIC`][fpic].  The
+latter results in the text section of executables being writable in some
+configurations, including common Linux configurations, subverting an important
+attack mitigation, and causing longer startup times by causing the linker to do
```
Comment (brson) [inline _posts/2017-02-09-Rust-1.15.1.md]: good spot

**Comment by brson** [comment]
Addressed.

**Comment by steveklabnik** [comment]
:shipit:

**Pull request merged**